zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,875 @@
+    return ds
+
+
+getdataset = op.attrgetter('dataset')
+dataset_name = op.attrgetter('name')
+
+
+def inline_novel_deltas(baseline, deltas, dates):
+    """Inline any deltas into the baseline set that would have changed our most
+    recently known value.
+
+    Parameters
+    ----------
+    baseline : pd.DataFrame
+        The first known values.
+    deltas : pd.DataFrame
+        Overwrites to the baseline data.
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+
+    Returns
+    -------
+    new_baseline : pd.DataFrame
+        The new baseline data with novel deltas inserted.
+    """
+    get_indexes = dates.searchsorted
+    return pd.concat(
+        (baseline,
+         deltas.loc[
+             (get_indexes(deltas[TS_FIELD_NAME].values, 'right') -
+              get_indexes(deltas[AD_FIELD_NAME].values, 'left')) <= 1
+         ].drop(AD_FIELD_NAME, 1)),
+        ignore_index=True,
+    )
+
+
+def overwrite_from_dates(asof, dense_dates, sparse_dates, asset_idx, value):
+    """Construct a `Float64Overwrite` with the correct
+    start and end date based on the asof date of the delta,
+    the dense_dates, and the dense_dates.
+
+    Parameters
+    ----------
+    asof : datetime
+        The asof date of the delta.
+    dense_dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    sparse_dates : pd.DatetimeIndex
+        The dates that appeared in the dataset.
+    asset_idx : int
+        The index of the asset in the block.
+    value : np.float64
+        The value to overwrite with.
+
+    Returns
+    -------
+    overwrite : Float64Overwrite
+        The overwrite that will apply the new value to the data.
+    """
+    first_row = dense_dates.searchsorted(asof)
+    last_row = dense_dates.get_loc(
+        sparse_dates[sparse_dates.searchsorted(asof) + 1],
+    ) - 1
+    if first_row > last_row:
+        return
+    yield Float64Overwrite(first_row, last_row, asset_idx, value)
+
+
+def adjustments_from_deltas_no_sids(dates,
+                                    dense_dates,
+                                    column_idx,
+                                    column_name,
+                                    assets,
+                                    deltas):
+    """Collect all the adjustments that occur in a dataset that does not
+    have a sid column.
+
+    Parameters
+    ----------
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    dense_dates : pd.DatetimeIndex
+        The dates that were in the dense data.
+    column_idx : int
+        The index of the column in the dataset.
+    column_name : str
+        The name of the column to compute deltas for.
+    deltas : pd.DataFrame
+        The overwrites that should be applied to the dataset.
+
+    Returns
+    -------
+    adjustments : dict[idx -> Float64Overwrite]
+        The adjustments dictionary to feed to the adjusted array.
+    """
+    ad_series = deltas.loc[:, AD_FIELD_NAME]
```

Comment:
```In [2]: %timeit df.loc[:, 'a']
The slowest run took 19.36 times longer than the fastest. This could mean that an intermediate result is being cached
10000 loops, best of 3: 17 µs per loop

In [3]: %timeit df['a']
The slowest run took 41.98 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 1.14 µs per loop
```